config/README.md

```diff
@@ -36,4 +36,34 @@ std.manifestYamlDoc(lib(cluster, coordinator))
 3. Run the following `jsonnet` command. This will generate a file called `generated.yaml` within the directory.
 ```
 	make config-gen
-```
\ No newline at end of file
+```
+
+## Configuration Schemas
+
+At this time, each service's configuration is defined as a Go struct in a 
+corresponding `config` package. For instance, `m3aggregator`'s config is 
+in [src/cmd/services/m3aggregator/config/config.go](https://github.com/m3db/m3/blob/master/src/cmd/services/m3aggregator/config/config.go).
+
+Validation is done using https://godoc.org/gopkg.in/go-playground/validator.v2 ;
+look for `validate` tags.
+
+## Advanced: Environment Variable Interpolation
+M3 uses [go.uber.org/config](https://godoc.org/go.uber.org/config) to load 
+its configuration. This means that we support environment variable interpolation
+of configs. For example:
+
+Given config of 
+
+```
+foo: ${MY_ENV_VAR}
+```
+
+and an environment of `MY_ENV_VAR=bar`, we will load:
```

Comment:
i made a few comments on your other PR, mind pulling those in here?

Reply:
Done! Sorry about the weirdness there; I think the branches are sorted now.